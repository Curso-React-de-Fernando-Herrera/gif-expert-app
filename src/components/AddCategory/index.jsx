import { useState } from "react"
import PropTypes from 'prop-types'

const CreateCategory = ({ setCategories }) => {
  const [newCategory, setNewCategory] = useState('')

  const handleChangeInput = e => setNewCategory(e.target.value)

  const handleAdd = e => {
    e.preventDefault()

    newCategory.trim() &&
      setCategories(lastCategories => [newCategory, ...lastCategories])
      
    setNewCategory('')
  }

  return <form onSubmit={handleAdd}>
    <input type="text" value={newCategory} onChange={handleChangeInput}/>
    <button>Agregar</button>
  </form>
}

CreateCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default CreateCategory
