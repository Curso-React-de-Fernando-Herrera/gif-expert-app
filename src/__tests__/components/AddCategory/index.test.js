import { shallow } from 'enzyme'
import AddCategory from 'components/AddCategory'

describe('<AddCategory /> component', () => {
  let setCategories = jest.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories} />)

  beforeEach(() => {
    setCategories = jest.fn()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })

  test('should be rendered', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should be simulate change input', () => {
    const value = 'Hello'
    const inputText = wrapper.find('input')
    
    inputText.simulate('change', { target: {value} })
    
    const returnedText = wrapper.find('input').props().value

    expect(returnedText).toBe(value)
  })

  test('should NOT submit when the text is empty', () => {
    const form = wrapper.find('form')
    form.simulate('submit', { preventDefault(){} })

    expect(setCategories).not.toHaveBeenCalled()
  })

  test('should be find gif and wish input', () => {
    const value = 'Dragon Ball'

    wrapper.find('input').simulate('change', { target: {value} })
    wrapper.find('form').simulate('submit', { preventDefault(){} }) 

    const inputValue = wrapper.find('input').props().value

    expect(setCategories).toHaveBeenCalled()
    expect(setCategories).toHaveBeenCalledTimes(1)
    expect(setCategories).toHaveBeenCalledWith( expect.any(Function) )

    expect(inputValue).toBe('')
    expect(inputValue).not.toBe(value)
  })
})