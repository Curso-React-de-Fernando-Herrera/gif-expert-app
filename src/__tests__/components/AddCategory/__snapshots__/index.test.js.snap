// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory /> component should be rendered 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button>
    Agregar
  </button>
</form>
`;
